custom/ilsgateway/api.py

```diff
@@ -549,9 +549,14 @@ def get_or_create_msd_zone(region):
             location.save()
 
             interface = SupplyInterface(self.domain)
-            if ilsgateway_location.type == 'FACILITY' and not interface.get_by_location(location):
-                interface.create_from_location(self.domain, location)
-                location.save()
+            if ilsgateway_location.type == 'FACILITY':
+                if not interface.get_by_location(location):
+                    interface.create_from_location(self.domain, location)
+                    location.save()
+                else:
+                    sql_location = location.sql_location
+                    if not sql_location.supply_point_id:
+                        location.save()
```

Comment:
Yep, supply point will be created in save method.

No, there was only about 5 locations affected with that problem. I have fixed this by manual resave on ILS domain.